Intersect.Client/Core/Fade.cs

```diff
@@ -21,28 +23,38 @@ public enum FadeType
 
         private static float sFadeAmt;
 
-        private static float sFadeRate = 3000f;
+        private static float sFadeRate = ClientConfiguration.DEFAULT_FADE_SPEED_MS;
 
         private static long sLastUpdate;
 
-        public static void FadeIn()
+        private static bool InformServer { get; set; }
+
+        public static void FadeIn(float speed, bool informServer = false)
         {
+            sFadeRate = speed;
             sCurrentAction = FadeType.In;
             sFadeAmt = 255f;
             sLastUpdate = Timing.Global.MillisecondsUtc;
+            InformServer = informServer;
         }
 
-        public static void FadeOut()
+        public static void FadeOut(float speed, bool informServer = false)
         {
+            sFadeRate = speed;
             sCurrentAction = FadeType.Out;
             sFadeAmt = 0f;
             sLastUpdate = Timing.Global.MillisecondsUtc;
+            InformServer = informServer;
         }
 
-        public static void Cancel()
+        public static void Cancel(bool informServer = false)
         {
             sCurrentAction = FadeType.None;
             sFadeAmt = default;
+            if (informServer)
```

Comment:
pass in `informServer` into `InformServerOfCompletion()` as a parameter called `force` and the check would be `if (force || InformServer)`

Reply:
Done, good call